dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py

```diff
@@ -1171,10 +1212,13 @@ def build_relations(self):
             relations_found = re.findall(r'RELATIONHYSTERESIS\(tmp[0-9]+, .*?, .*?, [0-9]+, .*?\);', transform_rawbody_to_string(eq.get_raw_body()))
             for relation in relations_found:
                 index_relation = relation.split(", ")[3]
-                if eq.get_is_diff_eq():
-                    map_relations[index_relation] = ["DIFF", eq.get_src_fct_name()]
-                else:
-                    map_relations[index_relation] = ["ALG", eq.get_src_fct_name()]
+                eq_type = ALGEBRAIC
+                if eq.get_type() == DIFFERENTIAL:
+                    eq_type = DIFFERENTIAL
+                elif eq.get_type() == MIXED:
+                    eq_type = self.find_equation_type_of_test(eq.get_body_for_setf(), relation.split(", ")[0].replace("RELATIONHYSTERESIS(",""), eq.get_evaluated_var())
```

Comment:
This might not be totally true (if condA x +0 else if condB der(x) = 0 else der(x) = 42), but rethinking about it I guess that you are right: we cannot guess which condition is going to be activated at a specific time step so we should systematically put ALGEBRAIC, is that ok with you?

Reply:
Fine for me with ALGEBRAIC.
It's the most secured default choice I believe.

Reply:
done